--- script.old.js	2026-01-20 16:07:18.347801401 +0100
+++ script.js	2026-01-20 15:15:56.441587104 +0100
@@ -1,12 +1,15 @@
+import { next, repeat, throwError } from "../../../utils/node.js";
+import { ArtWorks, FatalError } from "../utils.js";
+
 const CHECK_TASK_INTERVAL = 3000;
 const MAX_ATTEMPTS = 100;
 
-export async function costs({ env, inputs }) {
+export function costs({ env, inputs }) {
   if (env.scope.ARTWORKS_USER === "user") {
     return 0;
   }
 
-  const { batchSize, performance } = inputs;
+  const { batchSize = 1, performance = "speed" } = inputs;
 
   const { costs, details } = (() => {
     switch (performance) {
@@ -25,9 +28,6 @@
 }
 
 export async function run({ env, inputs, state }) {
-  const { throwError, repeat, next, download } = require("@piper/node");
-  const { ArtWorks, FatalError } = require("artworks");
-
   const { ARTWORKS_USER, ARTWORKS_PASSWORD } = env.variables;
   if (!ARTWORKS_USER) {
     throwError.fatal("Please, set ARTWORKS_USER in environment");
@@ -55,6 +55,7 @@
       batchSize,
       // SDXL
       styles,
+      templates,
     } = inputs;
 
     const payload = {
@@ -62,7 +63,15 @@
       isFast: true,
       payload: {
         base64: false,
-        prompt,
+        prompt: (() => {
+          const chunks = [prompt];
+          for (const t of templates || []) {
+            if (t.trainedWords?.length > 0) {
+              chunks.push(...t.trainedWords);
+            }
+          }
+          return chunks.join(", ");
+        })(),
         negativePrompt,
         checkpoint,
         cfgScale,
@@ -87,11 +96,17 @@
           }
         })(),
         sharpness,
+        ...(templates?.length > 0
+          ? {
+              loras: templates.map(({ fullFileName, weight }) => ({
+                modelName: fullFileName,
+                weight,
+              })),
+            }
+          : {}),
       },
     };
 
-    console.log(JSON.stringify(payload, null, 2));
-
     try {
       const task = await artworks.createTask(payload);
       console.log(`Task created ${task}`);
@@ -119,7 +134,9 @@
     if (attempt > MAX_ATTEMPTS) {
       try {
         await artworks.cancelTask(task);
-      } catch (e) {}
+      } catch (_e) {
+        // Ignore errors when canceling task
+      }
 
       const now = new Date();
       const time = (now - new Date(startedAt)) / 1000;
@@ -144,12 +161,10 @@
           },
         });
       }
-      let images = results.images.map((i) => i.url);
+      const images = results.images.map((i) => i.url);
       return next({
         outputs: {
-          images: (await Promise.all(images.map((url) => download(url)))).map(
-            ({ data }) => data
-          ),
+          images,
         },
         costs: costs({ env, inputs }),
       });
